File: internship_cs_mar23/jury_solutions/cpp/c.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
	int n, k;
	cin >> n >> k;
	vector<int> arr(n);
	for (int i = 0; i < n; ++i)
		cin >> arr[i];
	int l = 0, zeros = 0;
	long long sum = 0, ans = 0;
	for (int r = 0; r < n; ++r) {
		sum += arr[r];
		zeros += arr[r] == 0;
		while (sum > k || zeros > 1) {
			zeros -= arr[l] == 0;
			sum -= arr[l++];
		}
		if(sum <= k && zeros < 2)
			ans += r - l + 1;
	}
	cout << ans;
}

File: internship_cs_mar23/jury_solutions/python/c.py
n, k = map(int, input().split())
arr = list(map(int, input().split()))

l, zeros, sm, ans = 0, 0, 0, 0
for r in range(n):
    sm += arr[r]
    zeros += arr[r] == 0
    while sm > k or zeros > 1:
        zeros -= arr[l] == 0
        sm -= arr[l]
        l += 1
    if sm <= k and zeros < 2:
        ans += r - l + 1
print(ans)
